src/ex03_project/test_median.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Anders Mølmen Høst'
+__email__ = 'anderhos@nmbu.no'
+
+# INF200. Ex_03-Task B
+
+
+def median(data):
+    """
+    Returns median of data.
+
+    :param data: An iterable of containing numbers
+    :return: Median of data
+    """
+
+    sdata = sorted(data)
+    n = len(sdata)
+    if n == 0:
+        raise ValueError
+    return (sdata[n//2] if n % 2 == 1
+            else 0.5 * (sdata[n//2 - 1] + sdata[n//2]))
+
+
+# Testing if the function returns the correct value for a one-element list
+
+
+def test_single():
+    data1 = [1]
+    assert median(data1) == 1
+
+
+# Testing if the function returns the correct value for a odd numbered list
+
+
+def test_list_odd():
+    data2 = [1, 3, 5, 7, 9]
+    assert median(data2) == 5
+
+
+# Testing if the function returns the correct value for an even numbered list
+
+
+def test_list_even():
+    data3 = [2, 4, 6, 8, 10]
+    assert median(data3) == 6
+
+
+"""
+Testing if the function returns the correct value for an ordered numbered 
+list
+"""
+
+
+def test_list_ordered():
+    data4 = [1, 2, 3, 4, 5]
+    assert median(data4) == 3
+
+
+"""
+Testing if the function returns the correct value for a reversed 
+list
+"""
+
+
+def test_list_reverse():
+    data5 = [5, 4, 3, 2, 1]
+    assert median(data5) == 3
+
+
+"""
+Testing if the function returns the correct value for an unordered numbered 
+list
+"""
+
+
+def test_list_unordered():
+    data6 = [6, 9, 11, 2, 4]
+    assert median(data6) == 6
+
+
+# Testing the function for an empty list
+
+
+def test_list_empty():
```

Comment:
This test will fail, it will give a value error not returning a zero

Reply:
Thanks. Edited the code to make the test pass